Automattic/wp-calypso
# Reader: Reintroduce the Discover page

This PR is part of the project to reintroduce the discover page to Reader

<img width="705" alt="Screenshot 2023-06-16 at 16 42 40" src="https://github.com/Automattic/wp-calypso/assets/5560595/38beeed9-ce01-479e-9663-f81d438b373f">

We will keep the page private until we are complete by keeping the `isDiscoverEnabled` flag disabled.

I'm also limiting the scope of this initial PR to try to do the following;

- [x] Add Discover to side menu
- [x] Update Discover stream to use `read/tags/cards` endpoint
- [x] Add discover sidebar that lists recommended sites returned in `read/tags/cards` request

In a follow up PR we can look to add the horizontal menu above the discover post stream as per designs.
Some tasks I can think of...
* Pass recommended/related tags to a new horizontal tab menu - might be able to use the `READER_TAGS_RECEIVE` action
* Create horizontal tab component that renders tags based off action like `READER_TAGS_RECEIVE`

### Test
* Check with `isDiscoverEnabled` ([here](https://github.com/Automattic/wp-calypso/blob/de0756112aa5264e0761f47c9120d2b3789c682d/client/reader/discover/helper.js#L22)) disabled that the Discover menu isn't visible and the Following post streams (and others) work as expected.
* Enable the `isDiscoverEnabled` flag and confirm the discover menu item appear
* Confirm the Discover page loads as expected with a recommended post stream on the left and a popular sites sidebar on the right.
